src/services/refactors/moveToFile.ts

```diff
@@ -915,26 +912,91 @@ function getRangeToMove(context: RefactorContext): RangeToMove | undefined {
 
     const startNodeIndex = findIndex(statements, s => s.end > range.pos);
     if (startNodeIndex === -1) return undefined;
-
     const startStatement = statements[startNodeIndex];
-    if (isNamedDeclaration(startStatement) && startStatement.name && rangeContainsRange(startStatement.name, range)) {
-        return { toMove: [statements[startNodeIndex]], afterLast: statements[startNodeIndex + 1] };
-    }
 
     const overloadRangeToMove = getOverloadRangeToMove(file, startStatement);
-    if (overloadRangeToMove) {
-        return overloadRangeToMove;
+    /**function [|add(x: number, y: number): number;
+       function add(x: string, y: string): string;|]
+                        or
+       function [|add(x: number, y: number): number;
+       function add(x: string, y: string): string;
+       |]
+                        or
+       function [|add(x: number, y: number): number;
+       function add(x: string, y: string): string;
+       |]const a = 1;
+     */
+    if (overloadRangeToMove && (statements[overloadRangeToMove.end].end >= range.end || range.end === statements[overloadRangeToMove.end].end + 1)) {

```

Comment:
This `range.end === node.end + 1` stuff doesn’t feel right. Is this intended to capture the possibility of the range including a trailing newline? What if it includes a trailing space and a trailing newline, or two trailing newlines?

Reply:
Yes, it was for a trailing newline. I think this change should fix the other possibilities.